src/components/App.tsx

```diff
@@ -194,13 +195,22 @@ class App extends React.Component<any, AppState> {
   unmounted: boolean = false;
   actionManager: ActionManager;
 
-  public state: AppState = {
-    ...getDefaultAppState(),
-    isLoading: true,
+  public static defaultProps: Partial<ExcalidrawProps> = {
+    width: window.innerWidth,
+    height: window.innerHeight,
   };
 
   constructor(props: any) {
     super(props);
+    const defaultAppState = getDefaultAppState();
+
+    const { width, height } = props;
+    this.state = {
+      ...defaultAppState,
+      width,
+      height,
+    };
```

Comment:
why replacing `state` with indirect `defaultProps` which then must be spread in the constructor? Also, you got rid of the `isLoading: true` in the process.

I'd keep `public state`, and just add the width/height there. Were you solving some problem?

Reply:
I didn't get you, I am checking old width prop and new width prop when updating the width state, when do you think the state .width will not reflect the props.width coz componentDIdUpdate will always get triggered when prop updated ? could you explain more?

Reply:
nvm, I forgot how classical React works :).

Add the `isLoading: true` in the constructor and we can merge.